From 5e642c35ba6f4d037cbc01581150b765b8325bda Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sat, 26 Feb 2022 18:41:51 +0000
Subject: [PATCH] libstdc++: add allocator for iterator type in maglev::Graph

Otherwise std::allocator is used and there is no conversions
available.
---
 v8/src/maglev/maglev-graph.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/v8/src/maglev/maglev-graph.h b/v8/src/maglev/maglev-graph.h
index 60c9142..985cf40 100644
--- a/v8/src/maglev/maglev-graph.h
+++ b/v8/src/maglev/maglev-graph.h
@@ -8,14 +8,17 @@
 #include <vector>
 
 #include "src/maglev/maglev-basic-block.h"
+#include "src/zone/zone-allocator.h"
 
 namespace v8 {
 namespace internal {
 namespace maglev {
 
-using BlockConstIterator = std::vector<BasicBlock*>::const_iterator;
+using BlockConstIterator =
+    std::vector<BasicBlock*, ZoneAllocator<BasicBlock*>>::const_iterator;
 using BlockConstReverseIterator =
-    std::vector<BasicBlock*>::const_reverse_iterator;
+    std::vector<BasicBlock*,
+                ZoneAllocator<BasicBlock*>>::const_reverse_iterator;
 
 class Graph {
  public:
-- 
2.34.1

